engines/native/sandbox.go

```diff
@@ -35,6 +38,13 @@ func newSandbox(b *sandboxBuilder) (*sandbox, error) {
 		return nil, fmt.Errorf("Failed to temporary folder, error: %s", err)
 	}
 
+	if b.payload.Context != "" {
+		if err = fetchContext(b.payload.Context, homeFolder.Path()); err != nil {
```

Comment:
We actually have no such option. Since we download the context to the user home directory, we cannot start the download before the user is actually created. And we have to wait to the download finishing before start the process. Maybe moving the user creation to an earlier stage as well.